api/pkg/transformer/pipeline/compiler.go

```diff
@@ -172,14 +180,14 @@ func (c *Compiler) parseVariablesSpec(variables []*spec.Variable, compiledJsonPa
 		case *spec.Variable_Expression:
 			compiledExpression, err := c.compileExpression(v.Expression)
 			if err != nil {
-				return nil, nil
+				return nil, err
```

Comment:
Ah, was it a bug if we did not return error?

Reply:
yes, just realised this when adding new test